modules/nw-ovn-k-anp-best-practices.adoc

```diff
@@ -0,0 +1,61 @@
+
+* Selectors across multiple rules should not overlap. This ensures that that the same IPs do not appear in multiple policies, which can cause performance and scale limitations.
+
+* Avoid using `namedPorts` in conjunction with `PortNumber` and `PortRange` creates 6 ACLs and cause inefficiencies in your cluster.
+
+[id="considerations-using-banp_{context}"]
+== Considerations for using BaselineAdminNetworkPolicy
+
+* Only a single `BaselineAdminNetworkPolicy` (BANP) resource may be defined within a cluster, make sure you architect it thoughtfully. The following are supported uses for BANP that administrators might consider in designing their BANP:
+
+** You can set a default deny policy for cluster-local ingress in user namespaces. This BANP will make it so that users have to add `NetworkPolicy` to allow the ingress traffic that they want to allow, and if they do not add network policies for ingress it will be denied.
+
+** You can set a default deny policy for cluster-local egress in user namespaces. This BANP will make it so that users have to add `NetworkPolicy` to allow the egress traffic that they want to allow, and if they do not add network policies it will be denied.
+
+** You can set a default allow policy for egress to the in-cluster DNS service. Such a BANP will ensure that namespaced users do not have to set an allow egress `NetworkPolicy` to the in-cluster DNS service.
+
+** You can set an egress policy that allows internal egress traffic to all pods but denies access to all external endpoints (i.e `0.0.0.0/0` and `::/0`). This BANP will make it so user workloads can send traffic to other in-cluster endpoints, but not to external endpoints by default. `NetworkPolicy` can then be used by developers in order to allow their applications to send traffic to an explicit set of external services.
+
+* Be careful that you scope your BANP so that it only denies traffic in user namespaces and not to system namespaces. This is because the system namespaces do not have `NetworkPolicy` to override your BANP.
+
+[id="differences-with-netpol_{context}"]
+= Differences to consider between AdminNetworkPolicy and NetworkPolicy
+
+* Use explicit labels to reference your workloads within ANP and BANP rather than using the empty (`{}`) catch all selector to avoid accidental traffic selection.
+
+[IMPORTANT]
+====
+An empty namespace selector applied to a infrastructure namespace has the potential to leave your entire cluster unresponsive and in a non-functional state.
+====
+
+* API semantics for ANP are very different from `NetworkPolicy` objects. There is no implicit deny with ANP when you create the policy. You have to explicitly define what you need.
+
+* Unlike `NetworkPolicy` objects, `AdminNetworkPolicy` objects ingress rules are limited to in-cluster pods and namespaces so you cannot, and do not need to, set rules for ingress from the host network.
```

Comment:
@JoeAldinger Nothing stops for using for host networked pods, just the ANP rules are not enforced on them. They are only enforced pods created on cluster network.

Reply:
Rest of it looks good.
I did not see reference to ANP at same priority with non lapping rules.

Reply:
@palonsoro Are you good with wording of line number 61. If you are I will be OK to apply lgtm label.

Reply:
I worked with him over slack to develop that wording. I can message you the conversation if you'd like. He's out of office.

Reply:
Yes. The idea is not to mean that we can't have host networked pod. The idea is to make it clear that host networked pods are not impacted (i.e. they are not limited) by ANPs and BANPs.

If you believe that the wording leads to confusion in this regard, not sure if you could suggest some alternative. But to me, we can leave it as is.

Reply:
I _think_ `do not need to` gets at what we are trying to express.